app/recurring_expenses/views.py

```diff
@@ -39,53 +40,102 @@ def create_or_edit_recurring_expense(request, expense_id=None):
         edit_recurring_expense(request, expense_id=expense_id)
 
 
+_IMPORT_CONTENT_KEYS = {
+    "json": "application/json",
+    "csv": "text/csv",
+}
+
+_IMPORT_CONTENT_TYPES = {value: key for key, value in _IMPORT_CONTENT_KEYS.items()}
+
+
+def load_json_data(file):
+    try:
+        return json.loads(file.read())
+    except json.JSONDecodeError:
+        raise ValueError("Invalid JSON data in the file")
+
+
+def load_csv_data(file):
+    try:
+        f = file.read().decode("utf-8")
+        return csv.reader(f.splitlines(), delimiter=",")
+    except csv.Error:
+        raise ValueError("Invalid CSV data in the file.")
+
+
+def upload_json_date(data: dict):
+    for item in data:
+        try:
+            recurring_expense = RecurringExpense(
+                id=item["id"] if "id" in item else None,
+                active=item["active"],
+                particulars=item["particulars"],
+                amount=item["amount"],
+                currency=item["currency"],
+                period=item["period"],
+            )
+            recurring_expense.save()
+        except KeyError:
+            raise ValueError("Missing fields in the JSON data")
+
+
+def upload_csv_data(reader):
+    next(reader, None)
+    rows_imported = 0
+    rows_errored = 0
+    total_rows = 0
+
+    for row in reader:
+        if row[0].strip(" ").lower() == "totals":
+            break
+
+        total_rows += 1
+
+        try:
+            recurring_expense = RecurringExpense(
+                id=UUID(row[0]) if row[0] else None,
+                active=parse_bool(row[1]),
+                particulars=row[2].strip(" "),
+                amount=row[3].strip("$  "),
```

Comment:
I'm suprised this doesn't need to be converted to ``Decimal``?

Reply:
I mean, I tested your PR and it seems to work so 🤷

Reply:
I guess so, the csv upload logic was per-written I just refactored it into a different function. Guess it's fine?